extensions/vertx-web/deployment/src/main/java/io/quarkus/vertx/web/deployment/VertxWebProcessor.java

```diff
@@ -397,4 +428,84 @@ private static String dashify(String value) {
         }
         return ret.toString();
     }
+
+    private void detectConflictingRoutes(Map<RouteMatcher, MethodInfo> matchers) {
+        if (matchers.isEmpty()) {
+            return;
+        }
+        // First we need to group matchers that could potentially match the same request 
+        Set<LinkedHashSet<RouteMatcher>> groups = new HashSet<>();
+        for (Iterator<Entry<RouteMatcher, MethodInfo>> iterator = matchers.entrySet().iterator(); iterator.hasNext();) {
+            Entry<RouteMatcher, MethodInfo> entry = iterator.next();
+            LinkedHashSet<RouteMatcher> group = new LinkedHashSet<>();
+            group.add(entry.getKey());
+            matchers.entrySet().stream().filter(e -> {
+                if (e.getKey().equals(entry.getKey())) {
+                    // Skip - the same matcher
+                    return false;
+                }
+                if (e.getValue().equals(entry.getValue())) {
+                    // Skip - the same method
+                    return false;
+                }
+                if (e.getKey().getOrder() != entry.getKey().getOrder()) {
+                    // Skip - different order set
+                    return false;
+                }
+                return canMatchSameRequest(entry.getKey(), e.getKey());
+            }).map(Entry::getKey).forEach(group::add);
+            groups.add(group);
+        }
+        // Log a warning for any group that contains more than one member
+        boolean conflictExists = false;
+        for (Set<RouteMatcher> group : groups) {
+            if (group.size() > 1) {
+                Iterator<RouteMatcher> it = group.iterator();
+                RouteMatcher firstMatcher = it.next();
+                MethodInfo firstMethod = matchers.get(firstMatcher);
+                conflictExists = true;
+                StringBuilder conflictingRoutes = new StringBuilder();
+                while (it.hasNext()) {
+                    RouteMatcher rm = it.next();
+                    MethodInfo method = matchers.get(rm);
+                    conflictingRoutes.append("\n\t- ").append(method.declaringClass().name().toString()).append("#")
+                            .append(method.name()).append("()");
+                }
+                LOGGER.warnf(
+                        "Route %s#%s() can match the same request and has the same order [%s] as:%s",
+                        firstMethod.declaringClass().name(),
+                        firstMethod.name(), firstMatcher.getOrder(), conflictingRoutes);
+            }
+        }
+        if (conflictExists) {
+            LOGGER.warn("You can use @Route#order() to ensure the routes are not executed in random order");
```

Comment:
it's actually not random, it's the insertion order.

Reply:
Yes, but it's not guaranteed that we register the methods in the same order.